carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/router/HttpMethodInfo.java

```diff
@@ -72,39 +62,52 @@
         this.args = args;
     }
 
+    HttpMethodInfo(Method method, Object handler, HttpRequest request,
+                   HttpResponder responder, Object[] args,
+                   ExceptionHandler exceptionHandler, String mediaType,
+                   HttpStreaming httpStreaming) throws HandlerException {
+        this(method, handler, request, responder, args, exceptionHandler, mediaType);
+
+        if (!method.getReturnType().equals(Void.TYPE)) {
+            throw new HandlerException(HttpResponseStatus.INTERNAL_SERVER_ERROR,
+                    "Resource method should be void if it accepts chunked requests");
```

Comment:
On Sun, Jan 10, 2016 at 2:23 AM, Mohanadarshan Vivekanandalingam <
notifications@github.com> wrote:

> In
> carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/router/HttpMethodInfo.java
> <https://github.com/wso2/product-mss/pull/38#discussion_r49268116>:
>
> > @@ -72,39 +62,52 @@
> >          this.args = args;
> >      }
> >
> > +    HttpMethodInfo(Method method, Object handler, HttpRequest request,
> > +                   HttpResponder responder, Object[] args,
> > +                   ExceptionHandler exceptionHandler, String mediaType,
> > +                   HttpStreaming httpStreaming) throws HandlerException {
> > +        this(method, handler, request, responder, args, exceptionHandler, mediaType);
> > +
> > +        if (!method.getReturnType().equals(Void.TYPE)) {
> > +            throw new HandlerException(HttpResponseStatus.INTERNAL_SERVER_ERROR,
> > +                    "Resource method should be void if it accepts chunked requests");
>
> Better to have this like, if (!(Void.TYPE).equals(method.getReturnType()))
> to avoid possible NPE.
>

No that is incorrect. A method will always have a return type. If the
method has a null return type that should throw a NPE because there is an
error in the JDK.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wso2/product-mss/pull/38/files#r49268116>.
>



-- 
*Afkham Azeez*
Director of Architecture; WSO2, Inc.; http://wso2.com
Member; Apache Software Foundation; http://www.apache.org/
* <http://www.apache.org/>*
*email: **azeez@wso2.com* <azeez@wso2.com>
* cell: +94 77 3320919blog: **http://blog.afkham.org*
<http://blog.afkham.org>
*twitter: **http://twitter.com/afkham_azeez*
<http://twitter.com/afkham_azeez>
*linked-in: **http://lk.linkedin.com/in/afkhamazeez
<http://lk.linkedin.com/in/afkhamazeez>*

*Lean . Enterprise . Middleware*